### PR TITLE
Add Secrets Resource

### PR DESCRIPTION
**Pull request by nmanoogian**
Adds a Doppler resource for managing a single secret:

```
resource "doppler_secret" "apiKey" {
  project = "backend"
  config = "dev"
  name = "API_KEY"
  value = "My secret value"
}
```

### CODE REVIEW

File: doppler/api.go [hunk 1]
```diff
@@ -158,14 +158,71 @@ func (client APIClient) PerformRequest(req *http.Request, params []QueryParam) (
...
+	}
+	params = append(params, QueryParam{Key: "name", Value: secretName})
+	response, err := client.GetRequest(ctx, "/v3/configs/config/secret", params)
+	if err != nil {
+		return nil, err
+	}
+	var result Secret
+	jsonErr := json.Unmarshal(response.Body, &result)
+	if jsonErr != nil {
+		return nil, &APIError{Err: jsonErr, Message: "Unable to parse secrets"}
```
Comment (Piccirello) [inline doppler/api.go:196]: nit: `secrets` -> `secret`

File: doppler/resource_secret.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+		newSecret := RawSecret{Name: name, Value: &value}
+		err := client.UpdateSecrets(ctx, project, config, []RawSecret{newSecret})
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	d.SetId(getSecretId(project, config, name))
+
+	resourceSecretRead(ctx, d, m)
```
Comment (Piccirello) [inline doppler/resource_secret.go:67]: Should we do something with this function's return value?

**Review by Piccirello — APPROVED** [review]

File: doppler/resource_secret.go (hunk 2 above)
Comment (nmanoogian) [reply doppler/resource_secret.go:67]: Great catch! Terraform's example project doesn't even handle this

**Comment by nmanoogian** [comment]
@Piccirello Can I bug you to just look over that force push? During testing I noticed that our `computed` value wasn't getting recomputed properly because we were missing some custom diffing logic. I've also incorporated your suggested changed 👍

**Comment by Piccirello** [comment]
New changes look great 👍 Can you increase the length of the example password from 16 to 32 characters? Just on the off chance someone does use that code.

**Comment by nmanoogian** [comment]
> New changes look great 👍 Can you increase the length of the example password from 16 to 32 characters? Just on the off chance someone does use that code.

Sure thing 😄

**Pull request merged**